wpimath/src/main/java/edu/wpi/first/math/interpolation/Interpolatable.java

```diff
@@ -0,0 +1,25 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.math.interpolation;
+
+/**
+ * An object should extend interpolatable if you wish to linearly interpolate between a lower and
+ * upper bound, such as a robot position on the field between timesteps.
+ * @param <T> The class that is interpolatable.
+ */
+public interface Interpolatable<T> {
```

Comment:
Should be be LinearInterpolatable incase we want to add a SLERP interface in the future?

Reply:
Classes that subclass Interpolatable can define their own behavior for the interpolate function, including SLERP. I should change the javadoc to reflect this. Users can also pass a lambda to the buffer class that will be used to perform interpolation.

Reply:
Was this resolved?